contrib/debug-tools/openpaipdb/paipdb.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+"""
+Paipdb is a simple wrapper of the pdb.
+
+After Paipdb server starts, it will wait for client connect and all stdin/stdout will
+redicret to the client side.
+"""
+
+import os
+import sys
+import pdb
+import socket
+import time
+import traceback
+import threading
+import signal
+from functools import partial
+
+CURRENT_TAST_ROLE_NAME = os.environ.get("PAI_CURRENT_TASK_ROLE_NAME")
+TASK_ROLE_INDEX = os.environ.get("PAI_TASK_ROLE_INDEX")
+DEFAULT_ADDR = os.environ.get("PAI_HOST_IP_" + CURRENT_TAST_ROLE_NAME + "_" + TASK_ROLE_INDEX)
+DEBUG_PORT_NAME = os.environ.get("DEBUG_PORT_NAME", "")
+DEBUG_PORT_LIST = "PAI_PORT_LIST_" + CURRENT_TAST_ROLE_NAME + "_" + TASK_ROLE_INDEX + "_" + DEBUG_PORT_NAME
+DEFAULT_PORT = int(os.environ.get(DEBUG_PORT_LIST, "4444").split(',')[0])
```

Comment:
Is DEFAULT_PORT working? Or handle error when no debug port defines

Reply:
Thanks, will throw error and exit if no debug port here